src/app/games/lodestone/LodestoneRules.ts

```diff
@@ -0,0 +1,270 @@
+            }
+            return newPressurePlate;
+        } else {
+            return pressurePlate;
+        }
+    }
+    private removePressurePlate(board: LodestonePiece[][],
+                                start: Coord,
+                                direction: Direction,
+                                lodestones: LodestoneLodestonesPositions): void {
+        for (let coord: Coord = start; // eslint-disable-next-line indent
+             coord.isInRange(LodestoneState.SIZE, LodestoneState.SIZE); // eslint-disable-next-line indent
+             coord = coord.getNext(direction)) {
+            for (let player: number = 0; player < 2; player++) {
+                if (lodestones[player].equalsValue(coord)) {
+                    lodestones[player] = MGPOptional.empty();
+                }
+            }
+            board[coord.y][coord.x] = LodestonePieceNone.UNREACHABLE;
+        }
+    }
+    public isLegal(move: LodestoneMove, state: LodestoneState): MGPFallible<LodestoneInfos> {
+        const validityBeforeCaptures: MGPValidation = this.isLegalWithoutCaptures(state, move.coord, move.direction);
+        if (validityBeforeCaptures.isFailure()) {
+            return MGPFallible.failure(validityBeforeCaptures.getReason());
+        }
+
+        const infos: LodestoneInfos =
+            this.applyMoveWithoutPlacingCaptures(state, move.coord, move.direction, move.diagonal);
+        const numberOfCapturesInMove: number =
+            move.captures.top + move.captures.bottom + move.captures.left + move.captures.right;
+        const actualCaptures: number = Math.min(infos.captures.length, state.remainingSpaces());
+        if (numberOfCapturesInMove !== actualCaptures) {
+            return MGPFallible.failure(LodestoneFailure.MUST_PLACE_CAPTURES_ON_PRESSURE_PLATES());
+        }
+        for (const position of LodestonePressurePlate.POSITIONS) {
+            const pressurePlate: MGPOptional<LodestonePressurePlate> = state.pressurePlates[position];
+            if (pressurePlate.isPresent() && pressurePlate.get().remainingSpaces() < move.captures[position]) {
+                return MGPFallible.failure(LodestoneFailure.TOO_MANY_CAPTURES_ON_SAME_PRESSURE_PLATE());
+            }
+        }
+
+        return MGPFallible.success(infos);
+    }
+    public applyMoveWithoutPlacingCaptures(state: LodestoneState,
+                                           coord: Coord,
+                                           direction: LodestoneDirection,
+                                           diagonal: boolean,
+    ): LodestoneInfos {
+        let result: LodestoneInfos;
+        if (direction === 'pull') {
+            result = this.applyPull(state, coord, diagonal);
+        } else {
+            result = this.applyPush(state, coord, diagonal);
+        }
+        const previousLodestonePosition: MGPOptional<Coord> = state.lodestones[state.getCurrentPlayer().value];
```

Comment:
what about that map<player, thing> au lieu de [thing, thing] pattern qu'on avait entammé :p ?

Reply:
oui tout à fait monsieur !